File: OOYA_SP/Java/src/main/java/com/ran/splabjava/Developer.java
package com.ran.splabjava;

import java.util.List;

public class Developer extends Programmer {

    private String knowledgeArea;
    private String currentProject;

    public Developer(String name, String surname, Position position, List<String> skills,
                     String knowledgeArea, String currentProject) {
        super(name, surname, position, skills);
        this.knowledgeArea = knowledgeArea;
        this.currentProject = currentProject;
    }

    public String getKnowledgeArea() {
        return knowledgeArea;
    }

    public void setKnowledgeArea(String knowledgeArea) {
        this.knowledgeArea = knowledgeArea;
    }

    public String getCurrentProject() {
        return currentProject;
    }

    public void setCurrentProject(String currentProject) {
        this.currentProject = currentProject;
    }

    @Override
    void print() {
        System.out.println("Developer: " + getSurname() + " " + getName());
        System.out.println("Position: " + getPosition());
        System.out.println("Skills: " + getSkillsAsString());
        System.out.println("Knowledge area: " + getKnowledgeArea());
        System.out.println("Current project: " + getCurrentProject());
    }

}

File: OOYA_SP/C#/OPLab/Developer.cs
using System;
using System.IO;
using System.Collections.Generic;

namespace OPLab
{
    public class Developer : Programmer // Класс-наследник
	{
		// Приватные свойства класса наследника
		private String _knowledgeArea;
        private String _currentProject;


		public Developer(String name, String surname, Position position, List<String> skills,
                         String knowledgeArea, String currentProject) : base(name, surname, position, skills)
		{
			KnowledgeArea = knowledgeArea;
			CurrentProject = currentProject;
		}

        public string KnowledgeArea { get => _knowledgeArea; set => _knowledgeArea = value; } // Get- и Set-методы
        public string CurrentProject { get => _currentProject; set => _currentProject = value; }

         public override void  Print() // Переопределенный метод базового класса
		{
            Console.WriteLine("Developer: {0} {1}",this.Surname, this.Name );
            Console.WriteLine("Position: {0}", this.Position);
            Console.WriteLine("Skills: {0}", GetSkillsAsString());
            Console.WriteLine("Knowledge area: {0}", KnowledgeArea);
            Console.WriteLine("Current project: {0}", CurrentProject);
		    Console.WriteLine();
        }
    }
}